daemon/protocol/python/scoot/scoot.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+#This is the command line entry to the scoot daemon.
+#This code uses docopt to parse the command line.  docopt uses the  __doc__ string below 
+#for both parsing the command line into a dict and for displaying help.
+"""
+Usage:
+  scoot.py snapshot create <srcDir>
+  scoot.py snapshot checkout <snapshotId> <destDir>
+  scoot.py exec run <command> ... --snapshotId=<sid> [--timeout=<seconds>]
+  scoot.py exec poll <runId> ... [--wait=<wait>] [--all]
+  scoot.py exec abort <runId>
+  scoot.py echo <ping>
+
+  
+Submit commands to the Scoot daemon:
+
+snapshot commands:
+create   Creates a snapshot containing the files in the srcDir directory.  
+         This command returns the id for the snapshot.
+         
+checkout Copies the contents of snapshotId to destDir.  Files from snapshotId will overwrite files in destDir.
+
+
+execution commands:
+run             Copies the files from snapshotId (if a snapshotId is supplied), and runs the command.  
+                if timeout_ns is supplied, the daemon will wait timeout (seconds) for the command to complete.  
+                Default timeout is 0.5 seconds.
+
+poll            Get the status of runs.  RunIds must be a comma delimited list of run ids (1,3,5,6).
+                wait < 0 : poll will return the status values as soon as one or more run ids status is finished.
+                wait = 0 : poll will return the status values immediately
+                wait > 0 : poll will wait up to <wait> seconds for one of the run ids to reach finished state
+                If all is present, poll will return the status of all run ids, not just finished runs.
+
+abort           Abort the specified run.
+
+check daemon running:
+echo            The Scoot daemon echo's <ping> back to the client.  Use this command to verify that the daemon
+                is running.
+
+Options:
+  -h --help               Show this screen.
+  --snapshotId=<sid>      Install the snapshot with <sid> before running the command.
+  --timeout=<seconds>     Maximum time(seconds) to allow the command to run. Default 1 second.
+  --wait=<wait>           <0: wait indefinitely for at least one run to complete. 
+                          0:  return immediately with the status(s) of the runs.  
+                          >0: wait up to <wait>(seconds) for at least one of the runs to finish.  
+                          Default: 0.
+  --all                   Return the status of all the runs not just finished runs.
+
+"""
+import docopt
+import os
+import sys
+import re
```

Comment:
Just noticing this now, but we don't need the "re" module at all, we should remove it and replace all of the re.search statements with:
`if '<substring>' in str(e) or '<other_substring>' in str(e):`